### PR TITLE
add getter and setter for bpf program flags

### LINKED ISSUE #198: Get/set bpf program flags
Implement the ability to get and set program flags. This should be a wrapper around the libbpf library's

`__u32 bpf_program__flags(const struct bpf_program *prog);`

and

`int bpf_program__set_flags(struct bpf_program *prog, __u32 flags);`

(https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L8538-L8550). 

The libbpf-rs code for interacting with programs is located at libbpf-rs/src/program.rs

### PR DESCRIPTION
**Pull request by kckeiks**
Fixes #198

### CODE REVIEW

File: libbpf-rs/src/program.rs [hunk 1]
```diff
@@ -70,6 +70,11 @@ impl OpenProgram {
         };
         util::parse_ret(ret)
     }
+
+    pub fn set_prog_flags(&self, flags: u32) -> Result<()> {
```
Comment (chenhengqi) [inline libbpf-rs/src/program.rs:74]: Should be `set_flags` ?

Comment (kckeiks) [reply libbpf-rs/src/program.rs:74]: There is a method `set_prog_type` so I was following that pattern. I can change it though.

Comment (anakryiko) [reply libbpf-rs/src/program.rs]: good point, shouldn't prog_type and set_prog_type be just `type` and `set_type` to match libbpf's API? Though I suspect "type" is reserved keyword in Rust?

Comment (chenhengqi) [reply libbpf-rs/src/program.rs]: Yep, `type` is a keyword.

**Review by insearchoflosttime — APPROVED** [review]

**Comment by insearchoflosttime** [comment]
Thanks for adding this!

**Pull request merged**